go/vt/vtgate/planbuilder/dml.go

```diff
@@ -39,7 +39,17 @@ func buildUpdatePlan(upd *sqlparser.Update, vschema VSchema) (*engine.Route, err
 	route := &engine.Route{
 		Query: generateQuery(upd),
 	}
-	updateTable, _ := upd.Table.Expr.(sqlparser.TableName)
+	if len(upd.TableExprs) > 1 {
+		return nil, errors.New("unsupported: multi-table update")
+	}
+	aliased, ok := upd.TableExprs[0].(*sqlparser.AliasedTableExpr)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
+	updateTable, ok := aliased.Expr.(sqlparser.TableName)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
```

Comment:
I'm actually thinking we could be more permissive here. For example, we can convert `SelectEqualUnique` to `UpdateEqual`. So, instead of checking if keyspace is sharded, we can check route opcodes.
And by introducing a few more opcodes, we could perform multi-shard DMLs (in the future).